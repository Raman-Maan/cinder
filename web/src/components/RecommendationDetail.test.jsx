import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import RecommendationDetail from './RecommendationDetail';

describe('<RecommendationDetail />', () => {
  const testData = {
    userName: 'Abc Defg',
    userAge: 22,
    userGender: 'Male',
    userBio: 'I am test user with test bio',
    userPics:
      'http://markinternational.info/data/out/613/224320906-pic-of-animals.jpg'
  };

  it('renders without crashing', () => {
    // simple smoke test
    shallow(<RecommendationDetail />);
  });

  it('renders with empty props correctly', () => {
    const component = renderer.create(<RecommendationDetail />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders with props correctly', () => {
    const component = renderer
      .create(<RecommendationDetail userDetail={testData} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
