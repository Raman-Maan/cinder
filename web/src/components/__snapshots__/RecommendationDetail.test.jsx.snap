// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecommendationDetail /> renders with empty props correctly 1`] = `
<div
  className="UserDetailEmpty"
>
  <div
    className="empty"
  >
    <h3
      className="msg align-self-center"
    >
      There's no one new around you.
    </h3>
  </div>
</div>
`;

exports[`<RecommendationDetail /> renders with props correctly 1`] = `
<div
  className="UserDetailEmpty"
>
  <div
    className="empty"
  >
    <h3
      className="msg align-self-center"
    >
      There's no one new around you.
    </h3>
  </div>
</div>
`;
